spi/src/main/java/org/opensearch/jobscheduler/spi/JobExecutionContext.java

```diff
@@ -35,14 +38,33 @@ public JobExecutionContext(
         this.lockService = lockService;
         this.jobIndexName = jobIndexName;
         this.jobId = jobId;
+        this.accessToken = null;
+    }
+
+    public JobExecutionContext(
```

Comment:
Could we keep a single constructor since auth token can be passed in as null, or if you'd prefer Optional<AuthToken>?

Reply:
Done